src/System.Slices/src/System/MemoryUtils.Other.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    internal static partial class MemoryUtils
+    {
+        /// <summary>
+        /// platform independent fast memory comparison based on idea from String.EqualsHelper 
+        /// </summary>
+        internal static bool MemCmp<[Primitive]T>(Span<T> first, Span<T> second)
+            where T : struct
+        {
+            unsafe
+            {
+                // prevent GC from moving memory
+                var firstPinnedHandle = GCHandle.Alloc(first.Object, GCHandleType.Pinned);
+                var secondPinnedHandle = GCHandle.Alloc(second.Object, GCHandleType.Pinned);
+
+                try
+                {
+                    byte* firstPointer = (byte*)PtrUtils.ComputeAddress(first.Object, first.Offset).ToPointer();
+                    byte* secondPointer = (byte*)PtrUtils.ComputeAddress(second.Object, second.Offset).ToPointer();
+
+                    long bytesCount = first.Length * PtrUtils.SizeOf<T>();
+
```

Comment:
You should get something like
```C#
            int step = sizeof(void*) * 5;
            byte* firstPointerLimit = (byte*)(first.Length * PtrUtils.SizeOf<T>() - step);

            while (firstPointer < firstPointerLimit)
            {
                if (*((void**)firstPointer + 0) != *((void**)secondPointer + 0)) return false;
                if (*((void**)firstPointer + 1) != *((void**)secondPointer + 1)) return false;
                if (*((void**)firstPointer + 2) != *((void**)secondPointer + 2)) return false;
                if (*((void**)firstPointer + 3) != *((void**)secondPointer + 3)) return false;
                if (*((void**)firstPointer + 4) != *((void**)secondPointer + 4)) return false;

                firstPointer += step;
                secondPointer += step;
            }

            // check the tail bytes here

            return true;
```
It might work faster. I did not test this code though :-)